Update Model Optimizer User Guide (#10759)
openvinotoolkit/openvino: docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md

> Model Optimizer is a cross-platform command-line tool that facilitates the transition between the training and deployment environment, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices.

Reviewer: Can we rename Model Optimizer Developer Guide to Model Optimizer User Guide?

Also can we remove deprecated pages?

Author: Yes, we will definitely rename it.

Author: Done

> Model Optimizer is a cross-platform command-line tool that facilitates the transition between the training and deployment environment, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices.

Reviewer 2: ```suggestion
Model Optimizer is a cross-platform command-line tool that facilitates transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices.
```

Author: Done

> Model Optimizer process assumes you have a network model trained using supported deep learning frameworks: TensorFlow*, PyTorch*, ONNX*, PaddlePaddle*, MXNet*, Caffe*, Kaldi*. Model Optimizer produces an Intermediate Representation (IR) of the network, which can be inferred with the [OpenVINO™ Runtime](../OV_Runtime_UG/openvino_intro.md).

Reviewer 2: ```suggestion
The Model Optimizer process assumes you have a model trained using one of the supported deep learning frameworks: TensorFlow, PyTorch, ONNX, PaddlePaddle, MXNet, Caffe, or Kaldi. Model Optimizer produces an Intermediate Representation (IR) of the network, which can be inferred with [OpenVINO™ Runtime](../OV_Runtime_UG/openvino_intro.md).
```

Author: Done

> Model Optimizer process assumes you have a network model trained using supported deep learning frameworks: TensorFlow*, PyTorch*, ONNX*, PaddlePaddle*, MXNet*, Caffe*, Kaldi*. Model Optimizer produces an Intermediate Representation (IR) of the network, which can be inferred with the [OpenVINO™ Runtime](../OV_Runtime_UG/openvino_intro.md).

Author: Done

> > **NOTE**: Model Optimizer does not infer models. Model Optimizer is an offline tool that runs before the inference takes place.

Reviewer 2: maybe "offline tool designed to convert and adjust models, so that they provide optimal performance when inferred with OpenVINO Runtime" ... the note should probably be incorporated in the initial description to avoid repeating content.

> > **NOTE**: Model Optimizer does not infer models. Model Optimizer is an offline tool that runs before the inference takes place.

Author: Done

> **TIP**: You also can work with the Model Optimizer inside the OpenVINO™ [Deep Learning Workbench](https://docs.openvino.ai/latest/workbench_docs_Workbench_DG_Introduction.html) (DL Workbench).
> [DL Workbench](https://docs.openvino.ai/latest/workbench_docs_Workbench_DG_Introduction.html) is a web-based graphical environment that enables you to optimize, fine-tune, analyze, visualize, and compare performance of deep learning models.

## Run Model Optimizer

> To convert the model to the Intermediate Representation (IR), run Model Optimizer:

Author: Done

> To convert the model to the Intermediate Representation (IR), run Model Optimizer:

Author: Done

```sh
mo --input_model INPUT_MODEL
```

> Model Optimizer provides parameters `--input` and `--input_shape` to override original input shapes for model conversion.

Author: Done

> For more information about these parameters, please refer to [Setting Input Shapes](prepare_model/convert_model/Converting_Model.md).

Author: Done

> In order to cut off undesirable parts of a model such as unsupported operations and training sub-graphs,
> `--input` and `--output` parameters can be used to define new inputs and outputs of the converted model.
> For more description, please refer to [Cutting Off Parts of a Model](prepare_model/convert_model/Cutting_Model.md).

Reviewer 2: ```suggestion
To cut off unwanted parts of a model, such as unsupported operations and training sub-graphs,
the `--input` and `--output` parameters can be used, defining new inputs and outputs of the converted model.
For a more detailed description, refer to [Cutting Off Parts of a Model](prepare_model/convert_model/Cutting_Model.md).
```

Author: Done

> Also, Model Optimizer can insert additional input pre-processing sub-graphs into the converted model.

Author: Done

> For this, there exist `--mean_values`, `scales_values`, `--layout`, and other parameters described

Author: Done

> For this, there exist `--mean_values`, `scales_values`, `--layout`, and other parameters described
> in [Optimizing Preprocessing Computation](prepare_model/Additional_Optimizations.md).

Reviewer 2: ```suggestion
Parameters like `--mean_values`, `scales_values`, or `--layout` can be used for this purpose. For more options,  refer to [Optimizing Preprocessing Computation](prepare_model/Additional_Optimizations.md).
```

Author: Done

> in [Optimizing Preprocessing Computation](prepare_model/Additional_Optimizations.md).

Author: Done

> in [Optimizing Preprocessing Computation](prepare_model/Additional_Optimizations.md).

Author: Done

> Model Optimizer provides compression parameter `--data_type` to generate IR of `FP16` data type. For more details,

Reviewer 2: ```suggestion
Model Optimizer's  `--data_type` compression parameter allows to generate IR of the `FP16` data type. For more details,
```

> Model Optimizer provides compression parameter `--data_type` to generate IR of `FP16` data type. For more details,
> please refer to [Compression of a Model to FP16](prepare_model/FP16_Compression.md).

Reviewer 2: ```suggestion
Model Optimizer's  `--data_type` compression parameter allows to generate IR of the `FP16` data type. 
For more details, refer to [Compression of a Model to FP16](prepare_model/FP16_Compression.md).
```

Author: Done

To get the full list of conversion parameters available in Model Optimizer, run the following command:

```sh
mo --help
```

> ## Examples of CLI Commands

Author: I think we need basic examples for each FW.

> ## Examples of CLI Commands

Author: Need to itemize examples and add disclaimer about disconnected examples. Add a reference to MO Tutorials.

> ## Examples of CLI Commands

Author: Done

Launch the Model Optimizer for the TensorFlow* BERT model in binary protobuf format with three inputs and explicitly specify input shapes
where the batch size and the sequence length equal 2 and 30. For more information about TensorFlow* models conversion,

> please refer to [Converting a TensorFlow* Model](prepare_model/convert_model/Convert_Model_From_TensorFlow.md).

Author: Done

```sh
mo --input_model bert.pb --input input_mask,input_word_ids,input_type_ids --input_shape [2,30],[2,30],[2,30]
```

> Launch the Model Optimizer for the TensorFlow* MobileNet model in SavedModel format.

Author: Done

```sh
mo --saved_model_dir MobileNet
```

> Launch the Model Optimizer for the ONNX* OCR model and explicitly specify new output. For more information about ONNX* models conversion,
> please refer to [Converting a ONNX* Model](prepare_model/convert_model/Convert_Model_From_ONNX.md).
> Note that PyTorch* models must be exported to ONNX* format before its conversion into IR,
> follow details from [Converting a PyTorch* Model](prepare_model/convert_model/Convert_Model_From_PyTorch.md).

Reviewer 2: ```suggestion
Launch Model Optimizer for an ONNX OCR model and explicitly specify new output. For more information on ONNX model conversion,
refer to [Converting an ONNX Model](prepare_model/convert_model/Convert_Model_From_ONNX.md).
Note that PyTorch models must be exported to the ONNX format before conversion into IR.
More details can be found in [Converting a PyTorch Model](prepare_model/convert_model/Convert_Model_From_PyTorch.md).
```

Author: Done

```sh
mo --input_model ocr.onnx --output probabilities
```

> Launch the Model Optimizer for the PaddlePaddle* UNet model and apply mean-scale normalization to the input.
> For more information about ONNX* models conversion, please refer to
> [Converting a PaddlePaddle* Model](prepare_model/convert_model/Convert_Model_From_Paddle.md).

Reviewer 2: ```suggestion
Launch Model Optimizer for a PaddlePaddle UNet model and apply mean-scale normalization to the input.
For more information on ONNX model conversion, refer to
[Converting a PaddlePaddle Model](prepare_model/convert_model/Convert_Model_From_Paddle.md).
```

```sh

> mo --input_model unet.pdmodel --input data --mean_values data[123,117,104] --scale 255

Author: yes, all examples are more or less real.

> mo --input_model unet.pdmodel --input data --mean_values data[123,117,104] --scale 255

Author: Removed `--input`

```

> Launch the Model Optimizer for the MXNet* SSD Inception V3 model and specify input layout.
> For more information about MXNet* models conversion, please refer to [Converting a MXNet* Model](prepare_model/convert_model/Convert_Model_From_MxNet.md).

Reviewer 2: ```suggestion
Launch Model Optimizer for an MXNet SSD Inception V3 model and specify input layout.
For more information on MXNet model conversion, refer to [Converting an MXNet Model](prepare_model/convert_model/Convert_Model_From_MxNet.md).
```

Author: Done

Launch the Model Optimizer for the MXNet* SSD Inception V3 model and specify input layout.
For more information about MXNet* models conversion, please refer to [Converting a MXNet* Model](prepare_model/convert_model/Convert_Model_From_MxNet.md).

```sh
mo --input_model INPUT_MODEL
mo --input_model ssd_inception_v3-0000.params --layout NCHW
```

> **NOTE**: Some models require using additional arguments, such as `--input`, `--input_shape`, and `--output`. To learn about when you need to use these parameters, refer to [General Conversion Parameters](prepare_model/convert_model/Converting_Model.md).

To adjust the conversion process, you may use [general parameters](prepare_model/convert_model/Converting_Model.md) and framework-specific parameters for:
* [TensorFlow](prepare_model/convert_model/Convert_Model_From_TensorFlow.md)
* [ONNX](prepare_model/convert_model/Convert_Model_From_ONNX.md)
* [PyTorch](prepare_model/convert_model/Convert_Model_From_PyTorch.md)
* [PaddlePaddle](prepare_model/convert_model/Convert_Model_From_Paddle.md)
* [MXNet](prepare_model/convert_model/Convert_Model_From_MxNet.md)
* [Caffe](prepare_model/convert_model/Convert_Model_From_Caffe.md)

> * [Kaldi](prepare_model/convert_model/Convert_Model_From_Kaldi.md)

Author: We seemed to agree on having each link right after each example

> Launch the Model Optimizer for the Caffe* AlexNet model with reversed input channels order between RGB and BGR.

Reviewer 3: ```suggestion
Launch the Model Optimizer for the Caffe* AlexNet model with the reversed input channels order between RGB and BGR:
```

Author: Done

> For more information about Caffe* models conversion, please refer to [Converting a Caffe* Model](prepare_model/convert_model/Convert_Model_From_Caffe.md).

Reviewer 3: Move such "For more info" lines to after sample command-lines

> Launch the Model Optimizer for the Caffe* AlexNet model with reversed input channels order between RGB and BGR.
> For more information about Caffe* models conversion, please refer to [Converting a Caffe* Model](prepare_model/convert_model/Convert_Model_From_Caffe.md).

Reviewer 2: ```suggestion
Launch Model Optimizer for a Caffe AlexNet model with input channels in the RGB format, which requires reversing.
For more information on Caffe model conversion, please refer to [Converting a Caffe Model](prepare_model/convert_model/Convert_Model_From_Caffe.md).
```

Author: Done

> Launch the Model Optimizer for the Kaldi* LibriSpeech nnet2 model. For more information about Kaldi* models conversion,
> please refer to [Converting a Kaldi* Model](prepare_model/convert_model/Convert_Model_From_Kaldi.md).

Reviewer 2: ```suggestion
Launch Model Optimizer for a Kaldi LibriSpeech nnet2 model. For more information on Kaldi model conversion, 
refer to [Converting a Kaldi Model](prepare_model/convert_model/Convert_Model_From_Kaldi.md).
```

Author: Done

> ## Videos

Author: will discuss it separately. It looks outdated definitely.

> ## Videos

Author: Will remove

> ## Videos

Author: Done
